modules/obj/src/mesh.js

```diff
@@ -0,0 +1,536 @@
+    dst.push(src[b + 0], src[b + 1], src[b + 2]);
+    dst.push(src[c + 0], src[c + 1], src[c + 2]);
+  }
+
+  addVertexPoint(a) {
+    const src = this.vertices;
+    const dst = this.object.geometry.vertices;
+
+    dst.push(src[a + 0], src[a + 1], src[a + 2]);
+  }
+
+  addVertexLine(a) {
+    const src = this.vertices;
+    const dst = this.object.geometry.vertices;
+
+    dst.push(src[a + 0], src[a + 1], src[a + 2]);
+  }
+
+  addNormal(a, b, c) {
+    const src = this.normals;
+    const dst = this.object.geometry.normals;
+
+    dst.push(src[a + 0], src[a + 1], src[a + 2]);
+    dst.push(src[b + 0], src[b + 1], src[b + 2]);
+    dst.push(src[c + 0], src[c + 1], src[c + 2]);
+  }
+
+  addColor(a, b, c) {
+    const src = this.colors;
+    const dst = this.object.geometry.colors;
+
+    dst.push(src[a + 0], src[a + 1], src[a + 2]);
+    dst.push(src[b + 0], src[b + 1], src[b + 2]);
+    dst.push(src[c + 0], src[c + 1], src[c + 2]);
+  }
+
+  addUV(a, b, c) {
+    const src = this.uvs;
+    const dst = this.object.geometry.uvs;
+
+    dst.push(src[a + 0], src[a + 1]);
+    dst.push(src[b + 0], src[b + 1]);
+    dst.push(src[c + 0], src[c + 1]);
+  }
+
+  addUVLine(a) {
+    const src = this.uvs;
+    const dst = this.object.geometry.uvs;
+
+    dst.push(src[a + 0], src[a + 1]);
+  }
+
+  // eslint-disable-next-line max-params
+  addFace(a, b, c, ua, ub, uc, na, nb, nc) {
+    const vLen = this.vertices.length;
+
+    let ia = this.parseVertexIndex(a, vLen);
+    let ib = this.parseVertexIndex(b, vLen);
+    let ic = this.parseVertexIndex(c, vLen);
+
+    this.addVertex(ia, ib, ic);
+
+    if (ua !== undefined && ua !== '') {
+      const uvLen = this.uvs.length;
+      ia = this.parseUVIndex(ua, uvLen);
+      ib = this.parseUVIndex(ub, uvLen);
+      ic = this.parseUVIndex(uc, uvLen);
+      this.addUV(ia, ib, ic);
+    }
+
+    if (na !== undefined && na !== '') {
+      // Normals are many times the same. If so, skip function call and parseInt.
+      const nLen = this.normals.length;
+      ia = this.parseNormalIndex(na, nLen);
+
+      ib = na === nb ? ia : this.parseNormalIndex(nb, nLen);
+      ic = na === nc ? ia : this.parseNormalIndex(nc, nLen);
+
+      this.addNormal(ia, ib, ic);
+    }
+
+    if (this.colors.length > 0) {
+      this.addColor(ia, ib, ic);
+    }
+  }
+
+  addPointGeometry(vertices) {
+    this.object.geometry.type = 'Points';
+
+    const vLen = this.vertices.length;
+
+    for (const vertex of vertices) {
+      this.addVertexPoint(this.parseVertexIndex(vertex, vLen));
+    }
+  }
+
+  addLineGeometry(vertices, uvs) {
+    this.object.geometry.type = 'Line';
+
+    const vLen = this.vertices.length;
+    const uvLen = this.uvs.length;
+
+    for (const vertex of vertices) {
+      this.addVertexLine(this.parseVertexIndex(vertex, vLen));
+    }
+
+    for (const uv of uvs) {
+      this.addUVLine(this.parseUVIndex(uv, uvLen));
+    }
+  }
+}
+
+export default class Mesh {
```

Comment:
The `Mesh` name is a little confusing, when comparing to other loaders and also to the fact that we are returning another type of mesh and loaders.gl talks about the "mesh category" etc. 

My preference would be to  just turn this into a `parseOBJ` function that takes the constructor arguments directly (and kill the mesh class).

Should we also name the file accordinglyL `parse-obj.js`?